apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx

```diff
@@ -83,7 +84,14 @@ export const SwapCommitButton = memo(function SwapCommitButton({
   const showWrap = wrapType !== WrapType.NOT_APPLICABLE
   const [isRoutingSettingChange, resetRoutingSetting] = useRoutingSettingChanged()
   const slippageAdjustedAmounts = useSlippageAdjustedAmounts(trade)
-  const routerAddress = SWAP_ROUTER_ADDRESSES[trade?.inputAmount?.currency?.chainId]
+
+  const deadline = useTransactionDeadline()
+  const [statusWallchain, approvalAddressForWallchain, wallchainMasterInput] = useWallchainApi(trade, deadline)
+
+  const routerAddress =
+    statusWallchain === 'found'
+      ? approvalAddressForWallchain
```

Comment:
One more need to check, this approval address is dynamic from api. It'd be better hardcoded